tests/testthat/test-routes.R

```diff
@@ -0,0 +1,110 @@
+context("routes")
+
+test_that("URLs are properly redirected with app$redirect", {
+ library(dashHtmlComponents)
+
+ app <- Dash$new()
+
+ app$redirect("/foo", "/")
+ app$redirect("/bar/*", "/*")
+
+ app$layout(htmlDiv(
+  "Hello world!"
+  )
+ )
+
+ request_foo <- fiery::fake_request(
+   "http://127.0.0.1:8050/foo"
+ )
+
+ request_bar <- fiery::fake_request(
+   "http://127.0.0.1:8050/bar/foo"
+ )
+
+ # start up Dash briefly to load the routes
+ app$run_server(block=FALSE)
+ app$server$stop()
+
+ response_foo <- app$server$test_request(request_foo)
+ response_bar <- app$server$test_request(request_bar)
+
+ expect_equal(
+   response_foo$status,
+   301L
+  )
+
+ expect_equal(
+   response_foo$headers$Location,
+   "/"
+ )
+
+ expect_equal(
+   response_bar$status,
+   301L
+  )
+
+ expect_equal(
+   response_bar$headers$Location,
+   "/*"
```

Comment:
I thought the intent here was that `app$redirect("/bar/*", "/*")` would redirect `/bar/foo` to `/foo`?

Reply:
Yeah, this was my own failure to understand the header specification; sorry about that. From reading the MDN link and the RFC linked below, I don't think wildcards are supported within `Location`. The test has been revised in 4a7f2f9.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location
https://tools.ietf.org/html/rfc2616#page-135